src/commcare_cloud/commands/deploy.py

```diff
@@ -26,7 +26,10 @@ class Deploy(CommandBase):
             Skip the steps involved in recording and announcing the fact of the deploy.
         """),
         Argument('--commcare-branch', help="""
-            The name of the commcare-hq git branch to deploy. 
+            The name of the commcare-hq git branch to deploy. Mutually exclusive with commcare-hash.
+        """, default=None),
+        Argument('--commcare-hash', help="""
+            The hash of a commcare-hq commit to deploy. Mutually exclusive with commcare-branch.
```

Comment:
Thoughts on `--commcare-hash` versus `--commcare-commit`? Both `branch` and `commit` are deeply intertwined with our git repo, so it seems preferably to use the `git` terminology for both?

Reply:
+1 for `--commcare-commit` or if this supports any name for a given commit (branch, tag, commit) then `--commcare-rev`

Reply:
Ha, I went back and forth over naming. My read of [this article](https://www.atlassian.com/git/tutorials/refs-and-the-reflog?section=special-refs) was that both branches and hashes are ways to describe a commit, so it'd be odd to have both "branch" and "commit" args that do different things, and that "hash" is the best way to refer to the SHA. Is that not the case?

This doesn't support tags - [this section](https://github.com/dimagi/commcare-cloud/pull/3676/files#diff-ad8be748d06708f97faa6e0587d23784R142-R147) - would need changes, and I'd rather not do that unless/until there's a need.

Reply:
I agree and it was in the back of my mind that it is strange to have both `--commcare-branch` as well as this new option. Can you elaborate on how they are different? Is it ever the case that one would use both at the same time?

> This doesn't support tags - [this section](https://github.com/dimagi/commcare-cloud/pull/3676/files#diff-ad8be748d06708f97faa6e0587d23784R142-R147) would need changes...

I think `git rev-parse` (which takes a branch name, tag name, commit hash, etc. and returns a commit hash) might work nicely there. [`vellum-to-hq` does something similar](https://github.com/dimagi/commcare-hq/blob/5b79c82821bd9f306f461884771db5d5306dcdfc/scripts/vellum-to-hq#L201).

Reply:
Thank you, having `gitutils` to reference makes this simpler.